### PR TITLE
docs: creating the sidebar navigation docs

### PR DESCRIPTION
**Pull request by SiTaggart**
Add some docs for the navigation.

Make it easier to put a beta badge in a navigation item by controlling the layout for the consumer.

## Checklist

- [ ] Work in a "Draft" branch whilst you are getting feedback to reduce Chromatic costs. Once you are ready for approval, convert the PR to "Ready to review"
- [ ] Ensure all `required` checks have passed
- [ ] Icon changes must have a product design review
- [ ] Website content changes must have a product design review
- [ ] At least two engineers have reviewed any code changes
- [ ] At least one engineering lead has reviewed any core package changes or repository infrastructure
- [ ] Website visual regression testing is run by adding `🕵🏻‍♀️ Run website visual regression` label

### CODE REVIEW

File: packages/paste-core/components/sidebar/src/navigation/SidebarNavigationItem.tsx [hunk 1]
```diff
@@ -70,13 +70,19 @@ const SidebarNavigationItem = React.forwardRef<HTMLAnchorElement, SidebarNavigat
         aria-current={selected ? 'page' : undefined}
```
Comment (serifluous) [inline packages/paste-core/components/sidebar/src/navigation/SidebarNavigationItem.tsx:70]: todo: pls change the description in package.json to "Sidebar Navigation controls both flat and hierarchical navigation structures within an application's sidebar."

File: packages/paste-website/src/pages/components/sidebar-navigation/index.mdx [hunk 2]
```diff
@@ -0,0 +1,849 @@
+export const meta = {
+  title: 'Sidebar Navigation',
+  package: '@twilio-paste/sidebar',
+  description:
+    'Sidebar navigation comprises a set of user interface controls, designed to facilitate both flat and hierarchical navigation structures within an application.',
```
Comment (serifluous) [inline packages/paste-website/src/pages/components/sidebar-navigation/index.mdx:5]:
```suggestion
    'Sidebar Navigation controls both flat and hierarchical navigation structures within an application's sidebar.',
```

File: packages/paste-website/src/pages/components/sidebar-navigation/index.mdx [hunk 3]
```diff
@@ -0,0 +1,849 @@
...
+};
+
+<NormalizedComponentHeader
+  categoryRoute={SidebarCategoryRoutes.COMPONENTS}
+  githubUrl="https://github.com/twilio-labs/paste/tree/main/packages/paste-core/components/sidebar"
+  storybookUrl="?path=/story/components-sidebar-navigation--default"
+  description={meta.description}
+  data={props.data}
+  name="Sidebar Navigation"
+  description="Sidebar navigation comprises a set of user interface controls, designed to facilitate both flat and hierarchical navigation structures within an application."
```
Comment (serifluous) [inline packages/paste-website/src/pages/components/sidebar-navigation/index.mdx:80]:
```suggestion
  description="Sidebar Navigation controls both flat and hierarchical navigation structures within an application's sidebar."
```

File: packages/paste-website/src/pages/components/sidebar-navigation/index.mdx [hunk 4]
```diff
@@ -0,0 +1,849 @@
...
+</Box>`}
+</LivePreview>
+
+## Guidelines
+
+### About Sidebar Navigation
+
+Sidebar navigation comprises a set of user interface controls, designed to facilitate both flat and hierarchical navigation structures within an application. The navigation components are presentational and can be utilized to create navigation structures ranging from simple lists to complex, deeply nested structures.
+
+We recommend limiting the nesting depth to three levels. For navigation beyond this depth, consider also using the [In-page Navigation](/components/in-page-navigation).
```
Comment (serifluous) [inline packages/paste-website/src/pages/components/sidebar-navigation/index.mdx:192]:
```suggestion
Sidebar Navigation controls both flat and hierarchical navigation structures within an application's sidebar. The navigation components are presentational and can be used to create navigation structures ranging from simple lists to complex, deeply nested structures.

We recommend limiting the nesting depth to 3 levels. For navigation beyond this depth, consider also using the [In-page Navigation](/components/in-page-navigation).
```

File: packages/paste-website/src/pages/components/sidebar-navigation/index.mdx [hunk 5]
```diff
@@ -0,0 +1,849 @@
...
+
+### Accessibility
+
+Nested sidebar navigation employs the [Disclosure pattern](https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/).
```
Comment (serifluous) [inline packages/paste-website/src/pages/components/sidebar-navigation/index.mdx:196]:
```suggestion
Nested sidebar navigation uses the [ARIA Disclosure pattern](https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/).
```

File: packages/paste-website/src/pages/components/sidebar-navigation/index.mdx [hunk 6]
```diff
@@ -0,0 +1,849 @@
...
+- Every `SidebarNavigationItem` must implement an anchor element by assigning a valid URL to the `href` attribute.
+- A `selected` `SidebarNavigationItem` is designated as the "current page" via `aria-current="page"`.
+
+## Examples
+
+### Flat
+
+In its most basic form, the Sidebar Navigation can be utilized to create a straightforward, flat list of navigation items. This is ideal for smaller applications or single-product use cases.
+
+Each `SidebarNavigationItem` should be provided with an `href`, a valid URL that the user will navigate to upon interaction. This facilitates opening links in new tabs via cmd/ctrl + clicking or right-clicking on the item, ensuring proper keyboard navigation for keyboard-only users.
```
Comment (serifluous) [inline packages/paste-website/src/pages/components/sidebar-navigation/index.mdx:208]:
```suggestion
In its most basic form, Sidebar Navigation can be used to create a flat list of navigation items. This is ideal for smaller applications or single-product use cases.

Each `SidebarNavigationItem` should be provided with an `href`, a valid URL that the user will navigate to upon interaction. This facilitates opening links in new tabs via CMD/CTRL + clicking or right-clicking on the item, ensuring proper keyboard navigation for keyboard-only users.
```

File: packages/paste-website/src/pages/components/sidebar-navigation/index.mdx [hunk 7]
```diff
@@ -0,0 +1,849 @@
...
+        Insights
+      </SidebarNavigationItem>
+    </SidebarNavigation>
+  </SidebarBody>
+</Box>`}
+</LivePreview>
+
+### Hierarchical
+
+Sidebar Navigation also supports the creation of hierarchical information architectures using nested disclosures. You can establish "groups" of navigation sections by deploying the `SidebarNavigationSeparator` between groups of items.
```
Comment (serifluous) [inline packages/paste-website/src/pages/components/sidebar-navigation/index.mdx:277]:
```suggestion
Sidebar Navigation also supports hierarchical information architectures with nested navigation items. You can establish "groups" of navigation sections by using `SidebarNavigationSeparator` between groups of items.
```

Comment (serifluous) [reply packages/paste-website/src/pages/components/sidebar-navigation/index.mdx:277]: I think there's a simpler way to explain this Hierarchical section overall, but I'll just take a look over again when I reread through all the docs.

**Review by serifluous — CHANGES_REQUESTED** [review]

File: packages/paste-website/src/pages/components/sidebar-navigation/index.mdx [hunk 8]
```diff
@@ -0,0 +1,849 @@
...
+
+##### SidebarNavigationItem
+
+| Prop        | Type              | Description                                                                               | Default                     |
+| ----------- | ----------------- | ----------------------------------------------------------------------------------------- | --------------------------- |
+| `element?`  | `string`          | Overrides the default element name to apply unique styles with the Customization Provider | `'SIDEBAR_NAVIGATION_ITEM'` |
+| `children?` | `React.ReactNode` |                                                                                           |                             |
+| `href`      | `string`          | String for the url to link to.                                                            |                             |
+| `selected`  | `boolean`         | Set to indicate this item matches to the current page the user is on                      |                             |
+| `icon`      | `React.ReactNode` | Any product icons cna be used here                                                        |                             |
```
Comment (serifluous) [inline packages/paste-website/src/pages/components/sidebar-navigation/index.mdx:781]:
```suggestion
| `icon`      | `React.ReactNode` | Any product icons can be used here                                                        |                             |
```

File: packages/paste-website/src/pages/components/sidebar-navigation/index.mdx [hunk 9]
```diff
@@ -0,0 +1,849 @@
...
+
+##### SidebarNavigationDisclosure
+
+| Prop        | Type              | Description                                                                                                             | Default                           |
+| ----------- | ----------------- | ----------------------------------------------------------------------------------------------------------------------- | --------------------------------- |
+| `element?`  | `string`          | Overrides the default element name to apply unique styles with the Customization Provider                               | `'SIDEBAR_NAVIGATION_DISCLOSURE'` |
+| `children?` | `React.ReactNode` |                                                                                                                         |                                   |
+| `baseId?`   | `string`          | Base ID string to base subsequent, dynamic ids on                                                                       |                                   |
```
Comment (serifluous) [inline packages/paste-website/src/pages/components/sidebar-navigation/index.mdx:789]:
```suggestion
| `baseId?`   | `string`          | Base ID string to base subsequent, dynamic IDs on                                                                       |                                   |
```

File: packages/paste-core/components/sidebar/src/navigation/SidebarNavigationItem.tsx (hunk 1 above)
Comment (SiTaggart) [reply packages/paste-core/components/sidebar/src/navigation/SidebarNavigationItem.tsx:70]: Navigation doesn't have it's own package.json, it's shared with Sidebar container. Do you mean to update the very top line of the page?

Comment (serifluous) [reply packages/paste-core/components/sidebar/src/navigation/SidebarNavigationItem.tsx:70]: ya, that

File: packages/paste-website/src/pages/components/sidebar-navigation/index.mdx [hunk 10]
```diff
@@ -0,0 +1,849 @@
...
+| `element?` | `string` | Overrides the default element name to apply unique styles with the Customization Provider | `'SIDEBAR_NAVIGATION_SEPARATOR'` |
+
+##### useSidebarNavigationDisclosureState
+
+Pass these as part of an object to the hook.
+
+| Prop       | Type      | Description                                                   | Default |
+| ---------- | --------- | ------------------------------------------------------------- | ------- |
+| `baseId?`  | `string`  | Set to create a base ID that all dynamic IDs will be based on |         |
+| `visible?` | `boolean` | Set the initial visibilty of the Disclosure                   |         |
```
Comment (serifluous) [inline packages/paste-website/src/pages/components/sidebar-navigation/index.mdx:831]:
```suggestion
| `visible?` | `boolean` | Set the initial visibility of the Disclosure                   |         |
```

Comment (serifluous) [reply packages/paste-website/src/pages/components/sidebar-navigation/index.mdx:831]: caught one more typo, but I'll approve.

**Review by serifluous — APPROVED** [review]

File: packages/paste-website/src/pages/components/sidebar-navigation/index.mdx (hunk 10 above)
Comment (SiTaggart) [reply packages/paste-website/src/pages/components/sidebar-navigation/index.mdx:831]: Not much point in approving. When I accept it'll blow it away

Comment (serifluous) [reply packages/paste-website/src/pages/components/sidebar-navigation/index.mdx]: it makes me feel better

**Review by serifluous — APPROVED** [review]

**Review by TheSisb — APPROVED** [review]

**Pull request merged**